### ISSUE TITLE
Language chooser ?

### ISSUE THREAD

**Issue by bigrck64** [Dimillian/ACHNBrowserUI#44]
As you know, the http://acnhapi.com/ can display any catalog item in the correct localisation.
Would it be possible to add on option to let user choose the database language ?

**Comment by Dimillian** [comment]
For items we use a local catalog dump, which for now is english only. In the mid/long I think we'll need to localize it fully. For villagers I'll display their name in your local language if available quite soon as we have the info from acnhapi yeah :)
And for the rest of the app, the UI, anyone is welcome to do a PR do make it available :)

**Comment by ericlewis** [comment]
should be relatively easy to generate a localization from this: https://pastebin.com/B8N5KKsh.
has strings for diff languages in relation to items.

**Comment by vknabel** [comment]
What are the preferred plans for this?
Do you prefer translation relying on Localizable.strings or by patching the JSON-files?

**Comment by Dimillian** [comment]
I’ll take a look at that a bit later, but with 5000+ items we’ll use a translated data source, I’ll just pull another json.

**Comment by vknabel** [comment]
Maybe [imthe666st/ACNH](https://github.com/imthe666st/ACNH) can help you here. I also wrote a short [playground script to generate a Localizable.strings file](https://gist.github.com/vknabel/75ad69b3e6a26685cabe62f6ff22bc2c).

**Comment by Dimillian** [comment]
This is perfect! Thank you so much for the repo link. With those JSON we just have to match the internal id with our datasource and done. Crazy. I'll be able to do it in no time :)

**Comment by Dimillian** [comment]
Note: We'll not use localizable.string. I'll pull the Locale from the phone, and then load the appropriate translation JSON and replace the item name with the one pulled from the file.

**Comment by Dimillian** [comment]
You're code is a great help, I'm almost done :)

**Comment by Dimillian** [comment]
<img width="559" alt="Screenshot 2020-05-11 at 08 55 43" src="https://user-images.githubusercontent.com/535509/81532761-3a75cc80-9365-11ea-9a5e-877690ba4e21.png">

**Comment by vknabel** [comment]
Great to hear and happy to help!
The localizable approach was just the quickest and easiest way to get a prototype working. I guess patching the dataset is better, too!

As a side note: in the current data set, the first letter is capitalized, whereas in the linked one most names are lowercased in English.

**Comment by Dimillian** [comment]
Yup fixed that too, commiting in 5 minutes. You'll see, I like the approach I did. We load current localization table in memory in just query it from the Item.

**Comment by Dimillian** [comment]
https://github.com/Dimillian/ACHNBrowserUI/commit/081c8469d9f8b4c04ae336a49565967ce9f6ec01 @vknabel It's basic but it works :) So happy we could have that nailed down in so little time!

**Comment by Dimillian** [comment]
You're welcome to do adjustment as you see fit :)

**Comment by vknabel** [comment]
Thank you very much @Dimillian. I will take a look this evening in about 9-12 hours!

**Comment by bigrck64** [comment]
Awesome that was quick ! :)

**Comment by vknabel** [comment]
It works very well! From my perspective, this is done.
I opened #111 for a small and rare edge case. But good and blazingly fast work!

**Comment by Dimillian** [comment]
Fo items localizations data for now we have:

* English
* French
* Japanese
* Chinese
* German
* Russian
* Spanish
* Italian

**Comment by Arnoud-B** [comment]
> Would it be possible to add on option to let user choose the database language ?

That would be great indeed. I have my phone set to English but the game is set to French. Makes it quite hard to find items 😉

**Comment by TheVaan** [comment]
Hey @Arnoud-B, you already can do that. You can set App language to a different then device language. Go to iOS settings app, scroll down to app section, select AC Helper and set language to French.
(Screenshots are German, but I think you'll get what to do)
![image](https://user-images.githubusercontent.com/1108485/102050587-679ac980-3de3-11eb-8ca7-1b19566020cf.png) ![image](https://user-images.githubusercontent.com/1108485/102050599-6c5f7d80-3de3-11eb-978d-6ef7f27c369c.png)

**Comment by Arnoud-B** [comment]
Thanks @TheVaan. That solves it indeed, but I think most users won't go to the Settings app to find this option but instead rely on the app settings (in the app itself). Why not expose it there?

**Comment by TheVaan** [comment]
We did this because of apples localization guide lines. There are ways to build in a translation Switcher inside an app but this is often buggy (for SwiftUi). We decided to use the Apple way (introduced in iOS 13) and not to implement a buggy way. I agree it's not that easy to find but it's the way Apple wants us to do.

**Comment by Arnoud-B** [comment]
OK, good to know. Perhaps a mention in the app with a direct link to the AC Helper section in the Settings would be worth considering? (I've seen that in several apps)
